cmake/CLangFormat.cmake

```diff
@@ -0,0 +1,52 @@
+#  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 2 or (at your option)
+#  version 3 of the License.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+set(THIRD_PARTY_DIRS
+    zxcvbn/
+    http/qhttp/
+    streams/QtIOCompressor
+)
+
+set(THIRD_PARTY_FILES
```

Comment:
Include `core/ScreenLockListenerMac.h` and `core/ScreenLockListenerMac.cpp` here.

Reply:
Would be nice to have all the third party files centralised in one directory...

Reply:
I'm not sure those two are third party code, they need to be excluded because they contain Objective-C code.

Reply:
They are not third-party.

Maybe we can exclude them in another group? :thinking:

Reply:
I just renamed the lists so that we can exclude both third-party and obj-c files.